server/src/test/java/org/elasticsearch/cluster/allocation/FilteringAllocationIT.java

```diff
@@ -292,5 +294,90 @@ public void testTransientSettingsStillApplied() throws Exception {
                        String.join(", ", includeNodes)).isTrue();
         }
     }
+
+    /**
+     * https://github.com/crate/crate/issues/13529
+     */
+    @Test
+    public void test_allocate_table_to_cold_node_with_index_setting_overriding_cluster_settings() throws Exception {
```

Comment:
I see this "allow overriding cluster setting" feature more general. Personally, I'd prefer a simple unit test to verify the behaviour of the `FilterAllocationDecider` rather one more complex integration test with a specific scenario in mind.

Reply:
What do you think about using an `ESAllocationTestCase` to test the allocation behaviour of the `FilterAllocationDecider`?  This change is about changing the allocation behaviour based on input settings.

Reply:
> What do you think about using an ESAllocationTestCase to test the allocation behaviour of the FilterAllocationDecider? 

Yes, that's what I was thinking of.

Reply:
👍